From 811b529813a0b87ac1ba687ea1b9c2d9980975df Mon Sep 17 00:00:00 2001
From: Guille de la Cruz Dorado <guille.informatica.sevilla@gmail.com>
Date: Fri, 20 Mar 2015 23:11:04 +0100
Subject: [PATCH 2] [CURSO C] Error solucionado en main.c

Error en la variable campos para detectar parametros introducidos
correctamente.

Solucionado mediante el uso de flags.

Signed-off-by: Guille de la Cruz Dorado <guille.informatica.sevilla@gmail.com>
---
 ejercicios/2/src/main.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/ejercicios/2/src/main.c b/ejercicios/2/src/main.c
index 3440a46..d620d23 100644
--- a/ejercicios/2/src/main.c
+++ b/ejercicios/2/src/main.c
@@ -13,7 +13,7 @@ int main(int argc, char *argv[])
 	uint32_t id = 0;
 	char dueno[50];
 
-	int campos = 0;
+	uint8_t campos = 0;
 
 	static struct option long_options[] = {
                 { "matricula", required_argument, 0, 'm' },
@@ -30,19 +30,19 @@ int main(int argc, char *argv[])
         	switch(valor) {
         	case 'm':
 			strncpy(matricula, optarg, 14);
-			campos++;
+			campos |= 1 << 0;
                 	break;
         	case 'i':
 			id = atoi(optarg);
-			campos = campos + 2;
+			campos |= 1 << 1;
                 	break;
         	case 'g':
 			strncpy(marca, optarg, 49);
-			campos = campos + 3;
+			campos |= 1 << 2;
                 	break;
 		case 'd':
 			strncpy(dueno, optarg, 49);
-			campos = campos + 4;
+			campos |= 1 << 3;
 			break;
 		case 'h':
 			printf("Los parametros a usar son los siguientes:\n"
@@ -58,8 +58,8 @@ int main(int argc, char *argv[])
        		}
 	}
 
-	if (campos != 10) {
-		printf("Los campos instroducidos no son validos.\n");
+	if (campos != 15) {
+		printf("Los campos introducidos no son validos.\n");
 		return 0;
 	}
 
-- 
1.9.1

